src/main/java/Duke.java

```diff
@@ -1,10 +1,104 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for the Duke Class.
+     *
+     * @param filepath Path to file where the data for saved tasks are stored.
+     */
+    public Duke(String filepath) {
+        ui = new Ui();
+        storage = new Storage(filepath);
+        try {
+            tasks = new TaskList(storage.loadData());
+        } catch (FileNotFoundException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("data/duke.txt").run();
+    }
+
+    /**
+     * Runs the code for the Duke Assistant.
+     */
+    public void run() {
+        ui.printStartUp();
+        while (true) {
+            String in = ui.newInput();
+            Parser parser = new Parser(in);
+            if (parser.getCommand().equals("bye")) {
```

Comment:
Maybe you can abstract the parsing of commands inside the Parser class instead of putting it inside run()